### PR TITLE
fix: tradeConfirm render & composition

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

- Fixes a bug where we always render the `Flex` dom element to contain the trade warnings, even if there are none
- Refactors the `TradeConfirm` component to use better composition (the JSX is getting too hard to comprehend)

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

Small - mostly a refactor, with a small rendering logic change.

## Testing

- The confirm change should show no changes or regressions
- We should still render trade warnings, e.g. when trading into RUNE

### Engineering

☝️ 

### Operations

☝️ 

## Screenshots (if applicable)

Before:

Screenshot 2022-11-21 at 11 57 50 am

After:

Screenshot 2022-11-21 at 11 57 54 am

### CODE REVIEW

File: src/components/Trade/TradeConfirm/TradeConfirm.tsx [hunk 1]
```diff
@@ -200,12 +200,12 @@ export const TradeConfirm = () => {
     }
   }
 
-  const handleBack = () => {
+  const handleBack = useCallback(() => {
```
Comment (gomesalexandre) [inline src/components/Trade/TradeConfirm/TradeConfirm.tsx:203]: 💜

File: src/components/Trade/TradeConfirm/TradeConfirm.tsx [hunk 2]
```diff
@@ -237,22 +237,118 @@ export const TradeConfirm = () => {
...
+            <Stack spacing={0}>
+              <AlertDescription lineHeight='short'>
+                {translate('trade.intoRUNEBody')}
+              </AlertDescription>
+            </Stack>
+          </Alert>
+        )}
+      </Flex>
+    )
+    const shouldRenderWarnings = tradeWarningElement.props.children.type !== null
```
Comment (gomesalexandre) [inline src/components/Trade/TradeConfirm/TradeConfirm.tsx:286]: `tradeWarningElement.props.children.type` will always evaluate to undefined, we should check for the truthiness of at least one of the items of the `children` array

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
Refactor looking good and tight 🏅, tested it at runtime and saw no regressions.
Added one comment WRT one of the checks that looks like it will always return true (confirmed at runtime), will stamp after it is tackled and the PR is mergefixed

File: src/components/Trade/TradeConfirm/TradeConfirm.tsx (hunk 2 above)
Comment (0xApotheosis) [reply src/components/Trade/TradeConfirm/TradeConfirm.tsx]: Right you are, good catch!

Fixed in https://github.com/shapeshift/web/pull/3376/commits/341283cab62f9848ccc4af1455ae0148311413ae

**Review by gomesalexandre — APPROVED** [review]
LGTM now! 🥯

**Pull request merged**